pkg/filesystem/subvolume.go

```diff
@@ -185,7 +305,18 @@ func checkSnapshot(ctx context.Context, clientsets *k8sutil.Clientsets, operator
 
 // gets the list of subvolumegroup for the specified filesystem
 func getSubvolumeGroup(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace, fs string) ([]fsStruct, error) {
-	svgList, err := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolumegroup", "ls", fs, "--format", "json"}, operatorNamespace, clusterNamespace, true)
+	cmd := "ceph"
+	args := []string{"fs", "subvolumegroup", "ls", fs, "--format", "json"}
+
+	if checkForExternalStorage(ctx, clientsets, clusterNamespace) {
+		m, admin_id, admin_key := getExternalClusterDetails(ctx, clientsets, operatorNamespace, clusterNamespace)
+		cmd = "/bin/sh"
+		args = []string{
+			"-c",
+			fmt.Sprintf(`export CEPH_ARGS="-m=%s --id=%s --key=%s" && ceph fs subvolumegroup ls %s --conf=/var/lib/rook/%s/%s.config --format json`, m, admin_id, admin_key, fs, clusterNamespace, clusterNamespace),
+		}
+	}
+	svgList, err := exec.RunCommandInOperatorPod(ctx, clientsets, cmd, args, operatorNamespace, clusterNamespace, true)
```

Comment:
we should have helper function to run the command , it will become maintenance problem and easy to introduce bugs if we follow if/else approach to run same comment but with different ceph clusters.